crates/solver/src/main.rs

```diff
@@ -543,25 +543,34 @@ async fn main() {
             .await
             .map_err(|err| tracing::error!("Couldn't fetch market makable token list: {}", err))
             .ok();
-    let submission_nodes = args
+    let submission_nodes_with_url = args
         .transaction_submission_nodes
         .into_iter()
         .enumerate()
         .map(|(index, url)| {
             let transport = create_instrumented_transport(
-                HttpTransport::new(client.clone(), url, index.to_string()),
+                HttpTransport::new(client.clone(), url.clone(), index.to_string()),
                 metrics.clone(),
             );
-            web3::Web3::new(transport)
+            (web3::Web3::new(transport), url)
         })
         .collect::<Vec<_>>();
-    for node in &submission_nodes {
-        let node_network_id = node.net().version().await.unwrap();
+    for (node, url) in &submission_nodes_with_url {
+        let node_network_id = node.net().version().await.unwrap_or_else(|_| {
+            panic!(
+                "Unable to retrieve network id on startup using the submission node at {}",
+                url
+            )
```

Comment:
Interesting that this also prints the underlying error. I would not have expected that.

Reply:
Good catch, it doesn't! I used `.expect` at the beginning, which does print the error, but then directed by `clippy` I changed this to an `unwrap_or_else(|_| panic!(` and didn't notice that the error was gone. Now I explicitly added the error and updated the description.

Reply:
Now I'm confused. I specifically tried the code to see because I was skeptical but it worked. Now I'm not sure why it worked on my system. 😕 

Alternatively you can also use the `anyhow::Context` trait:
```rust
        let node_network_id = node
            .net()
            .version()
            .await
            .context(format!(
                "Unable to retrieve network id on startup using the submission node at {url}"
            ))
            .unwrap();
```

Reply:
If I undo the new changes I don't see the error:

```
2022-06-16T10:39:29.464Z ERROR shared::tracing: thread 'main' panicked at 'Unable to retrieve network id on startup using the submission node at http://bad.node/bad/path', crates/solver/src/main.rs:560:13:
thread 'main' panicked at 'Unable to retrieve network id on startup using the submission node at http://bad.node/bad/path', crates/solver/src/main.rs:560:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I did push the `.expect` version before creating the PR at some point, maybe you fetched the code at this point and switched to that branch instead?

Anyway, `context` makes sense so now I changed to that (and `{url}`!).

Reply:
Really splitting hairs though - makes no difference in the grand scheme of things.

Reply:
A nice overview of the ways to error out. :smile:

Reply:
🤣